src/GameObservers/GameObservers.cpp

```diff
@@ -0,0 +1,213 @@
+#include "GameObservers.h"
+
+// Observer definitions
+Observer::Observer(std::string uid) : id(new std::string(uid)) {}
+
+Observer::~Observer()
+{
+    if (id)
+    {
+        delete id;
+        id = NULL;
+    }
+}
+
+std::string Observer::getId()
+{
+    return *id;
+}
+
+// PhaseObserver definitions
+PhaseObserver::PhaseObserver(std::string uid) : Observer::Observer(uid)
+{
+    currentPlayer = new std::string();
+    action = new std::string();
+}
+
+PhaseObserver::~PhaseObserver()
+{
+    if (currentPlayer)
+    {
+        delete currentPlayer;
+        currentPlayer = NULL;
+    }
+
+    if (action)
+    {
+        delete action;
+        action = NULL;
+    }
+}
+
+void PhaseObserver::update(const GameState &state, std::string playerName, std::string playerAction)
+{
+    // make action lowercase and remove underscores
+    for (int i = 0; i < playerAction.length(); i++)
+    {
+        if (playerAction[i] == '_')
+        {
+            playerAction[i] = ' ';
+        }
+        else if (playerAction[i] >= 'A' || playerAction[i] <= 'Z')
+        {
+            playerAction[i] = std::tolower(playerAction[i]);
+        }
+    }
+
+    *currentPlayer = playerName;
+    *action = playerAction;
+    std::cout << "== Game Phase Information ==" << std::endl;
+    std::cout << "It is " << *currentPlayer << "'s turn: " << *action << std::endl;
+
+    std::cout << "Here is what the map now looks like:" << std::endl;
+    state.map->printNodes();
+}
+
+// StatisticsObserver definitions
+StatisticsObserver::StatisticsObserver(std::string uid) : Observer::Observer(uid){};
+
+std::string StatisticsObserver::graphRow(std::string header, int width, int value)
+{
+    // Start with the header of the row
+    std::string row = header;
+
+    // Add enough spaces so that all rows are equal length
+    for (int i = 0; i < (width - header.length()); i++)
+    {
+        row += " ";
+    }
+
+    // Add a symbol to display the end of the row
+    row += "|";
+
+    // Build the bar in the bar graph
+    for (int i = 0; i < value; i++)
+    {
+        row += "0";
```

Comment:
`row += "\u25A1";  // for white box
 row += "\u25A0";  // for black box`

I find this unicode also works, if you'd like to use

Reply:
I think these could be cooler than the zero! I'm on board.

Reply:
This is a great idea! Updated.